### PR TITLE
feat: Add new e2e test cases for example to create workspaces and start and…

### PR DESCRIPTION
**Pull request by flacatus**
This PR add new e2e test cases to chectl for minikube and minishift:
Test added:
- server:stop
- workspace:create
- workspace list

Related issue:
[https://github.com/eclipse/che/issues/16622]()

Signed-off-by: flacatus <flacatus@redhat.com>

### CODE REVIEW

File: test/e2e/util/devfile-example.yaml [hunk 1]
```diff
@@ -0,0 +1,67 @@
+metadata:
+  name: quarkus-tests
+projects:
+  - name: quarkus-quickstarts
```
Comment (tolusha) [inline test/e2e/util/devfile-example.yaml:4]: Use the very basic devfile.
No plugins, no commands

Comment (flacatus) [reply test/e2e/util/devfile-example.yaml:4]: ok

**Comment by flacatus** [comment]
test

**Comment by flacatus** [comment]
[test]

File: test/e2e/minikube.test.ts [hunk 2]
```diff
@@ -9,13 +9,18 @@
...
 import { expect, test } from '@oclif/test'
 
+import { E2eHelper } from './util/e2e'
+
+const helper = new E2eHelper()
 jest.setTimeout(600000)
 
-describe('e2e test', () => {
-  describe('server:start without parameters', () => {
+let execution_platform = 'kubernetes'
```
Comment (mmorhun) [inline test/e2e/minikube.test.ts:17]: I think we can use `const` here.

File: test/e2e/minikube.test.ts [hunk 3]
```diff
@@ -24,11 +29,44 @@ describe('e2e test', () => {
...
-      .stdout()
+      .it('Obtain access_token from keycloak.', async () => {
+        const token = await helper.Access_Token(execution_platform)
+        process.env.CHE_ACCESS_TOKEN = token
+      })
+  })
+})
+
+describe('Workspace creation, list, start, inject, delete. Support stop and delete commands for Eclipse Che server', () => {
+  const wait = (ms = 10) => new Promise(resolve => setTimeout(resolve, ms))
```
Comment (mmorhun) [inline test/e2e/minikube.test.ts:40]: Maybe we can move this to some utils?

File: test/e2e/minikube.test.ts [hunk 4]
```diff
@@ -24,11 +29,44 @@ describe('e2e test', () => {
...
+  // !TODO ADD coverage for next commands: workspace:delete and workspace:inject
+  describe('Create Workspace', () => {
+    test
+      .stdout({ print: true })
+      .command(['workspace:create', '--start', '--devfile=test/e2e/util/devfile-example.yaml'])
+      .exit(0)
+      .it('Create a workspace', async () => {
+        await wait(160000)
+        const workspace = await helper.WorkspaceID(execution_platform)
+        expect(workspace[0].status).to.contain('RUNNING')
```
Comment (mmorhun) [inline test/e2e/minikube.test.ts:51]: If workspace list is empty test will fail with unclear error. Let's check size first.

File: test/e2e/minishift.test.ts [hunk 5]
```diff
@@ -9,12 +9,17 @@
...
 import { expect, test } from '@oclif/test'
 
+import { E2eHelper } from './util/e2e'
+
+const helper = new E2eHelper()
 jest.setTimeout(600000)
 
-describe('e2e test', () => {
-  describe('server:start without parameters', () => {
+const execution_platform = 'openshift'
```
Comment (mmorhun) [inline test/e2e/minishift.test.ts:17]: To me just `PLATFORM` looks better. But up to you.

File: test/e2e/minishift.test.ts [hunk 6]
```diff
@@ -24,13 +29,46 @@ describe('e2e test', () => {
...
-      .stdout()
+      .it('Obtain access_token from keycloak and set it like environment variable.', async () => {
+        const token = await helper.Access_Token(execution_platform)
+        process.env.CHE_ACCESS_TOKEN = token
+      })
+  })
+})
+
+describe('Workspace creation, list, start, inject, delete. Support stop and delete commands for Eclipse Che server', () => {
+  const wait = (ms = 10) => new Promise(resolve => setTimeout(resolve, ms))
```
Comment (mmorhun) [inline test/e2e/minishift.test.ts:40]: This function duplicates code.

File: test/e2e/util/e2e.ts [hunk 7]
```diff
@@ -0,0 +1,107 @@
...
+  protected kubeHelper: KubeHelper
+  protected che: CheHelper
+  protected oc: OpenShiftHelper
+
+  private readonly axios: AxiosInstance
+  constructor() {
+    this.kubeHelper = new KubeHelper({})
+    this.che = new CheHelper({})
+    this.oc = new OpenShiftHelper()
+    const httpsAgent = new Agent({ rejectUnauthorized: false })
```
Comment (mmorhun) [inline test/e2e/util/e2e.ts:30]: What is the reason to have `rejectUnauthorized` turned off?

File: test/e2e/util/e2e.ts [hunk 8]
```diff
@@ -0,0 +1,107 @@
...
+  // Return `access_token` from OC/K8s. Receive the platform where che is deployed
+  async Access_Token(platform: string): Promise<string> {
+    const params = {
+      client_id: 'che-public',
+      username: 'admin',
+      password: 'admin',
+      grant_type: 'password'
+    }
+    try {
+      if (platform === 'openshift') {
```
Comment (mmorhun) [inline test/e2e/util/e2e.ts:46]: Is it possible to use `isOpenshiftPlatformFamily`?

File: test/e2e/util/e2e.ts [hunk 9]
```diff
@@ -0,0 +1,107 @@
...
+
+        return accessToken.data.access_token
+      }
+    } catch (error) {
+      return error
+    }
+  }
+
+  //Return all workspaces
+  async WorkspaceID(platform: string): Promise<any> {
```
Comment (mmorhun) [inline test/e2e/util/e2e.ts:65]: Method name usually should contain a verb. It's not clear what it does from this name.
`getAllWorkspaces(...)` WDYT ?

File: yarn.lock [hunk 10]
```diff
@@ -1525,7 +1525,7 @@ ecc-jsbn@~0.1.1:
 
 "eclipse-che@git://github.com/eclipse/che#master":
   version "0.0.0"
-  resolved "git://github.com/eclipse/che#417eb0347356124eb636e979e8bed48b491694e8"
+  resolved "git://github.com/eclipse/che#8c9740f6f593977cfc514629bcb2e06287536d6e"
```
Comment (mmorhun) [inline yarn.lock:1528]: Does this change really needed?

File: test/e2e/util/e2e.ts (hunk 7 above)
Comment (flacatus) [reply test/e2e/util/e2e.ts:30]: Is just for test purpose.

File: yarn.lock (hunk 10 above)
Comment (flacatus) [reply yarn.lock:1528]: was changed automatically, IDK the reason.

File: test/e2e/minikube.test.ts (hunk 3 above)
Comment (AndrienkoAleksandr) [reply test/e2e/minikube.test.ts:40]: For non tests we are using:
```
await cli.wait()
```

File: test/e2e/minikube.test.ts [hunk 11]
```diff
@@ -24,11 +29,98 @@ describe('e2e test', () => {
...
+    test
+      .stdout({ print: true })
+      .command(['workspace:create', '--devfile=test/e2e/util/devfile-example.yaml'])
+      .exit(0)
+      .it('Create a workspace and wait to be started')
+  })
+
+  describe('Start Workspace', () => {
+    it('Start a workspace using execa library', async () => {
+      const workspace_id = await helper.GetWorkspaceId(PLATFORM)
```
Comment (mmorhun) [inline test/e2e/minikube.test.ts:52]: `workspace_id` -> `workspaceId`
`GetWorkspaceId` -> `getWorkspaceId`

File: test/e2e/minishift.test.ts (hunk 5 above)
Comment (flacatus) [reply test/e2e/minishift.test.ts]: I like more PLATFORM!

File: test/e2e/minishift.test.ts [hunk 12]
```diff
@@ -24,15 +30,100 @@ describe('e2e test', () => {
           .and.to.contain('Command server:start has completed successfully')
       })
     test
-      .skip()
-      .stdout()
+      .it('Obtain access_token from keycloak and set it like environment variable.', async () => {
+        const token = await helper.Access_Token(PLATFORM)
```
Comment (mmorhun) [inline test/e2e/minishift.test.ts:34]: `Access_Token` -> `getAccessToken` or `retrieveAccessToken`

**Pull request merged**